plugins/sse-streaming/main.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+/*
+
+the following configuration must be present in the krakend.json file:
+
+"extra_config": {
+    "plugin/http-server": {
+      "name": ["sse-streaming"],
+      "sse-streaming": {
+        "endpoint": "/sse/{id}",
+        "backend_url_pattern": "/events-stream/{id}",
+        "backend_host": "http://localhost:9081"
+      }
+    }
+  }
+
+the provided endpoint overwrites any existing endpoint in the configuration file.
+
+*/
+
+// pluginName is the name of the plugin, used as a key in the configuration map.
+var pluginName = "sse-streaming"
+
+// HandlerRegisterer is the symbol the plugin loader will try to load. It must implement the Registerer interface.
+var HandlerRegisterer = registerer(pluginName)
+
+type registerer string
+
+// RegisterHandlers registers the handler function with the given name.
+func (r registerer) RegisterHandlers(f func(
+	name string,
+	handler func(context.Context, map[string]interface{}, http.Handler) (http.Handler, error),
+)) {
+	f(string(r), r.registerHandlers)
+}
+
+// registerHandlers extracts configuration and sets up the HTTP handler.
+func (r registerer) registerHandlers(ctx context.Context, extra map[string]interface{}, h http.Handler) (http.Handler, error) {
+	config, ok := extra[pluginName].(map[string]interface{})
+	if !ok {
+		return h, errors.New("configuration not found")
+	}
+
+	// Extract configuration values
+	endpoint, endpointOk := config["endpoint"].(string)
+	backendURLPattern, backendURLPatternOk := config["backend_url_pattern"].(string)
+	backendHost, backendHostOk := config["backend_host"].(string)
+
+	// Check if all required configuration values are present
+	if !endpointOk || !backendURLPatternOk || !backendHostOk {
+		return h, errors.New("missing required configuration values")
+	}
+
+	// Basic sanity checks on the configuration values
+	if endpoint == "" {
+		return h, errors.New("endpoint cannot be empty")
+	}
+	if backendURLPattern == "" {
+		return h, errors.New("backend_url_pattern cannot be empty")
+	}
+	if backendHost == "" {
+		return h, errors.New("backend_host cannot be empty")
+	}
+
+	// Validate the backend host
+	if _, err := url.ParseRequestURI(backendHost); err != nil {
+		return h, errors.New("invalid backend_host URL")
+	}
+
+	// Return a new HTTP handler that wraps the original handler with custom logic.
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// TODO: Performance optimize matchStrings; critical, every request to the API gateway uses this.
+		matchPaths, id := matchStrings(endpoint, req.URL.Path)
+
+		if !matchPaths {
+			h.ServeHTTP(w, req)
+			return
+		}
+
+		// Construct serverURL using the extracted ID
+		serverURL := fmt.Sprintf("%s%s", backendHost, strings.Replace(backendURLPattern, "{id}", id, 1))
```

Comment:
```suggestion
		serverURL := fmt.Sprintf("http://%s%s", backendHost, strings.Replace(backendURLPattern, "{id}", id, 1))
```
I just tried, I guess we need to add `http` here.